executor/runtime/docker/docker.go

```diff
@@ -109,6 +111,11 @@ var NoEntrypointError = &runtimeTypes.BadEntryPointError{Reason: errors.New("Ima
 // applications shall tolerate the presence of such names.
 var environmentVariableKeyRegexp = regexp.MustCompile("^[A-Za-z_][A-Za-z0-9_]*$")
 
+// Spectatord tag keys and values are only allowed to use characters in the set -._A-Za-z0-9.
+// Others will be converted to an _.
+// See https://github.com/Netflix-Skunkworks/spectatord#allowed-characters.
+var spectatordUnexpectedTagCharRegexp = regexp.MustCompile("[^-._A-Za-z0-9]")
```

Comment:
The current implementation replaces every illegal char to a `_`. Consecutive illegal chars become consecutive `_`s. If we want to replace consecutive illegal chars to a single `_`, I can tweak the expression to do so.

Reply:
I think that is probably fine.
It is not like a user can put arbitrary strings here in the first place.

Reply:
Multiple replacements is the correct implementation - this is how other Atlas clients handle it.